src/tests/unit/workload/test_search.py

```diff
@@ -42,97 +46,545 @@ def cbcsdk_mock(monkeypatch, cb):
 
 
 # ==================================== UNIT TESTS BELOW ====================================
+
 def test_compute_resource_by_id(cbcsdk_mock):
-    """Tests a simple compute resource querry"""
-    cbcsdk_mock.mock_request("GET", "/lcm/view/v1/orgs/test/compute_resources/1539610",
+    """Tests a simple compute resource query"""
+    cbcsdk_mock.mock_request("GET", "/lcm/view/v2/orgs/test/compute_resources/15396109?deployment_type=WORKLOAD",
                              FETCH_COMPUTE_RESOURCE_BY_ID_RESP)
     api = cbcsdk_mock.api
     resource = ComputeResource(api, "15396109")
     assert resource._model_unique_id == "15396109"
 
 
-def test_search_facet_async(cbcsdk_mock):
-    """Tests running the async_querry."""
+def test_aws_resource_by_id(cbcsdk_mock):
+    """Tests an AWS compute resource query."""
+    cbcsdk_mock.mock_request("GET", "/lcm/view/v2/orgs/test/compute_resources/7001?deployment_type=AWS",
+                             FETCH_AWS_RESOURCE_BY_ID_RESP)
+    api = cbcsdk_mock.api
+    resource = AWSComputeResource(api, "7001")
+    assert resource._model_unique_id == "7001"
+
+
+def test_search_resource_async(cbcsdk_mock):
+    """Tests running the async_query."""
     def post_validate(url, body, **kwargs):
         crits = body['criteria']
+        assert crits['deployment_type'] == ['WORKLOAD']
         assert crits['uuid'] == ['502277cc-0aa9-80b0-9ac8-6f540c11edaf']
+        assert body['sort'] == [{"field": "name", "order": "ASC"}]
 
-        return SEARCH_AND_FACET_COMPUTE_RESEOURCES
+        return SEARCH_COMPUTE_RESOURCES
 
-    cbcsdk_mock.mock_request("POST", "/lcm/view/v1/orgs/test/compute_resources/_search",
+    cbcsdk_mock.mock_request("POST", "/lcm/view/v2/orgs/test/compute_resources/_search",
                              post_validate)
     api = cbcsdk_mock.api
-    query = api.select(ComputeResource).set_uuid(['502277cc-0aa9-80b0-9ac8-6f540c11edaf'])
+    query = api.select(ComputeResource).set_uuid(['502277cc-0aa9-80b0-9ac8-6f540c11edaf']).sort_by("name", "ASC")
```

Comment:
Why are we sorting and getting a resource by id?

Reply:
Does `api.select(ComputeResource, "502277cc-0aa9-80b0-9ac8-6f540c11edaf")` work too get the resource by id as well?

Reply:
The call to `sort_by` is required to ensure that the sort criteria parameters are passed through OK to the request.  Since `sort_by` is handled by `BaseComputeResourceQuery`, we can test it in the "simple" test rather than the "bells and whistles" one.

Reply:
The tests should replicate expected behaviors our developers may perform. As much as we want to combine tests to reduce how much we need to write and maintain, there needs to be a balance of normal behavior too.